### PR TITLE
remove support for 'x' in version constraints

### PR DESCRIPTION
**Pull request by dimbleby**
removes the most bizarre of the many ways in which poetry tries to handle nonsensical version constraints

fixes https://github.com/python-poetry/poetry/issues/5990

### CODE REVIEW

File: src/poetry/core/constraints/version/patterns.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
     rf"^~=\s*(?P<version>{VERSION_PATTERN})$", re.VERBOSE | re.IGNORECASE
 )
 X_CONSTRAINT = re.compile(
```
Comment (Secrus) [inline src/poetry/core/constraints/version/patterns.py:19]: Isn't this whole pattern only for `x` in version constraints? Didn't dive too deep into the code, but on quick glance, just changing it still leaves some code related to "x constraint"

Comment (dimbleby) [reply src/poetry/core/constraints/version/patterns.py:19]: The code uses the term "x constraint" for some things with `*` in them, but so far as I know this is the only place that is allowing a literal `x`

**Comment by radoering** [comment]
I think I am fine with the change since it is an undocumented feature.

I thought about deprecating it first because it will hurt some people for sure, especially since it is in the build backend. However, a deprecation is not that simple in this case.

@Secrus Feel free to merge it if you agree to remove it.

File: src/poetry/core/constraints/version/patterns.py (hunk 1 above)
Comment (Secrus) [reply src/poetry/core/constraints/version/patterns.py:19]: Ah, okay, didn't notice the `*` in the old regex. My bad

**Review by Secrus — APPROVED** [review]

**Pull request merged**